### PR TITLE
feat(meth): emit Bismark-compatible XR/XG/XM auxiliary tags

### PR DESCRIPTION
**Pull request by nh13**
## Summary

`bwa-mem3 mem --meth` now emits Bismark `XR:Z` (read conversion), `XG:Z` (genome strand), and `XM:Z` (methylation call string) so the output BAM works directly with `bismark_methylation_extractor`, methylKit, methtuple, DMRfinder, epialleleR (and continues to work with MethylDackel and biscuit). The bwameth-style `YS:Z`/`YC:Z`/`YD:Z` tags are dropped from output (kept internally for SEQ restoration). `-V` `XR:Z` is suppressed under `--meth` to avoid the collision.

## How

The doubled c2t pac (`f-<chr>` + `r-<chr>`) is folded once at startup into an in-memory un-converted reference via a 5-row `(f, r) → original` recovery table. No on-disk format change. Per mapped record, `meth_build_xm` slices the recovered ref at the alignment footprint and walks BAM CIGAR jointly over SEQ and the ref window per Bismark's `methylation_call` ladder. Strand pick is driven by `XG:Z`, not by SAM `0x10`, so CTOT/OB reads are handled correctly.

## Test plan

- [ ] CI unit tests green.
- [ ] `test/meth/test.sh` Layers 1+2+3 pass.
- [ ] `test/meth/test_bismark_tags.sh` (new holodeck-driven golden-truth diff) passes — currently 179627/179627 XM-match on records with matching alignment, 0 XR/XG mismatches.
- [ ] Manual: `MethylDackel extract` and `bismark_methylation_extractor` run cleanly on a sorted `--meth` BAM.

## Summary by CodeRabbit

* **New Features**
  * mem --meth now emits Bismark-compatible tags XR:Z, XG:Z and XM:Z and restores original pre-conversion SEQ.
  * New opt-in chimera QC flag (--chimera-qc) implementing the longest-match <44% heuristic.

* **Bug Fixes**
  * Suppress conflicting XR:Z reference annotation under --meth to avoid tag collisions.

* **Documentation**
  * Comprehensive docs updated for XR/XG/XM tags, chimera QC behavior, and workflow guidance.

* **Tests**
  * Added unit and end-to-end tests validating XR/XG/XM emission, XM construction, and chimera-QC behavior.

[Review Change Stack](https://app.coderabbit.ai/change-stack/fg-labs/bwa-mem3/pull/90)

### CODE REVIEW

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review

File: src/bntseq.cpp [hunk 1]
```diff
@@ -411,65 +412,94 @@ int bns_intv2rid(const bntseq_t *bns, int64_t rb, int64_t re)
...
+{
+	if (ref_id) *ref_id = bns_pos2rid(bns, pos_f);
+	cnt_ambi_ctx ctx = { 0 };
+	bns_iter_ambi(bns, pos_f, len, cnt_ambi_visit, &ctx);
+	return ctx.nn;
+}
+
+void bns_get_seq_into(int64_t l_pac, const uint8_t *pac,
+                      int64_t beg, int64_t end,
+                      uint8_t *dst, int64_t *len_out)
```
[the remark this answers is not in the document]
Comment (nh13) [reply src/bntseq.cpp:469]: Declining this refactor — leaving the thread open for visibility but not implementing.

Rationale: the three production callers (`bwa.cpp:276`, `bwamem.cpp:447`, `bwamem_pair.cpp:211`) all size their thread-local `kvec_t(uint8_t)` scratch via `u8vec_scratch.h::ensure(rlen)` immediately before the `bns_*_into` call, and the parity test (`test/unit/test_bns_get_seq_parity.cpp`) round-trips both `_into` and `_v2` across the boundary cases (forward-only, reverse-only, bridging, swap-path, randomized, and now empty-post-clamp). Adding `int64_t dst_cap` would thread a fifth parameter through both helpers + 3 call sites + 5 test cases + the `u8vec_scratch` wrapper for a guard the existing scratch contract already provides at the call site.

Companion: the related empty-post-clamp regression flagged on the same hunk (PRRT_kwDOSGcDic6A8ADK) is fixed and now landed on both `bns_get_seq_into` and `bns_get_seq_v2` (siblings audit). That removes the only realistic way `*len_out` could become negative and poison downstream scratch sizing.

Happy to revisit if a future caller adds a new path that does not own the scratch sizing.

**Comment by nh13** [comment]
@coderabbitai review
